daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -841,11 +841,69 @@ func (devices *DeviceSet) createRegisterDevice(hash string) (*devInfo, error) {
 	return info, nil
 }
 
-func (devices *DeviceSet) createRegisterSnapDevice(hash string, baseInfo *devInfo, size uint64) error {
-	if err := devices.poolHasFreeSpace(); err != nil {
+func (devices *DeviceSet) takeSnapshot(hash string, baseInfo *devInfo, size uint64) error {
+	var (
+		doSuspend      bool
+		devinfo        *devicemapper.Info
+		deferredRemove = true
+		err            error
```

Comment:
Yes it would work, but that would scope `devinfo` within the if condition. We need to use `devinfo` later outside the if condition. So we need to declare both the variables {error, devinfo}. 

Golang has scoped variables, and it's painful :)

Shishir